core/src/main/java/org/apache/iceberg/SnapshotScan.java

```diff
@@ -66,7 +66,8 @@ protected Long snapshotId() {
 
   protected abstract CloseableIterable<T> doPlanFiles();
 
-  protected ScanMetrics scanMetrics() {
+  @Override
+  public ScanMetrics scanMetrics() {
```

Comment:
`ScanMetrics` shouldn't be exposed and should be treated as internal information as they can change while the scan is going on. What you'd rather want to do is to look at the `ScanReport`, which will be created once a scan is complete.

So in your case what would make sense is to have a custom `MetricsReporter` that receives the `ScanReport` when a scan is done, which you can then use to hook it up to the Spark UI.
In order to

Reply:
I believe this no longer applies too and the PR has been updated to support custom reporters.